packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../../../primitives/utils.dart';
+import '../../../../shared/common_widgets.dart';
+import '../../../../shared/table.dart';
+import '../../../../shared/table_data.dart';
+import '../../../../shared/utils.dart';
+import '../../memory_controller.dart';
+import 'allocation_profile_tracing_view_controller.dart';
+
+/// The default width for columns containing *mostly* numeric data (e.g.,
+/// instances, memory).
+const _defaultNumberFieldWidth = 80.0;
+
+class _TraceCheckBoxColumn extends ColumnData<TracedClass>
+    implements ColumnRenderer<TracedClass> {
+  _TraceCheckBoxColumn()
+      : super(
+          'Trace',
+          titleTooltip:
+              'Enable or disable allocation tracing for a specific type',
+          fixedWidthPx: scaleByFontFactor(55.0),
+          alignment: ColumnAlignment.left,
+        );
+
+  @override
+  bool get supportsSorting => false;
+
+  @override
+  Widget build(
+    BuildContext context,
+    TracedClass item, {
+    bool isRowSelected = false,
+    VoidCallback? onPressed,
+  }) {
+    final controller =
+        Provider.of<MemoryController>(context).allocationTracingController;
+    return Checkbox(
+      value: controller.isAllocationTracingEnabledForClass(item.cls),
+      onChanged: (value) async {
+        await controller.setAllocationTracingForClass(item.cls, value!);
+      },
+    );
+  }
+
+  @override
+  bool? getValue(TracedClass _) {
+    return null;
+  }
+
+  @override
+  int compare(TracedClass a, TracedClass b) {
+    return a.traceAllocations.boolCompare(b.traceAllocations);
+  }
+}
+
+class _AllocationTracingTableClassName extends ColumnData<TracedClass> {
+  _AllocationTracingTableClassName() : super.wide('Class');
+
+  @override
+  String? getValue(TracedClass stats) => stats.cls.name;
+
+  @override
+  bool get supportsSorting => true;
+}
+
+class _AllocationTracingTableInstances extends ColumnData<TracedClass> {
+  _AllocationTracingTableInstances()
+      : super(
+          'Instances',
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  int getValue(TracedClass dataObject) {
+    return dataObject.instances;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class AllocationTracingTable extends StatefulWidget {
+  const AllocationTracingTable({required this.controller});
+
+  final AllocationProfileTracingViewController controller;
+
+  @override
+  State<AllocationTracingTable> createState() => _AllocationTracingTableState();
+}
+
+class _AllocationTracingTableState extends State<AllocationTracingTable> {
+  SortDirection sortDirection = SortDirection.ascending;
+  ColumnData<TracedClass> secondarySortColumn = _classNameColumn;
```

Comment:
nit: sometimes it is helpful to have an `initState` method do the initialization so that it is clear what is stateful about this widget. At first glance, I looked at this and thought, "we aren't using any lifecycle methods besides `build`, so why does this need to be stateful"? And then looking closer I saw the call to `setState`. If these were late and then initialized in `initState` that would be more immediately clear.